### DISCUSSION TITLE
Keep track if connection to streaming API is still working

### DISCUSSION THREAD

**Discussion by seism0saurus** [andregasser/bigbone#429] [Q&A] [answered]
Hi,

I'm working on a social wall project for Mastodon with the bigbone library in the Java backend.
It connects to the streaming API of Mastodon via a `client.streaming().hashtag()` call.
Since the returned Closable from Java does not provide any methods to check the status I wonder, how I can detect if the connection is still healthy or needs to be restarted.
Is there any event, that is sent over the API in case of connection errors?

To my understanding an unhealthy connection should trigger some events, that are passed to my callback `onEvent()`-method, right?
https://github.com/andregasser/bigbone/blob/8b342097307afb6bd2e4fcc48a9490bd12f2cfab/bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt#L625

So where should be no unhealthy connection without a trigger?

kind regards
seism0saurus

**Comment by PattaFeuFeu** [comment, marked as answer]
Hi!

> Is there any event, that is sent over the API in case of connection errors?
> To my understanding an unhealthy connection should trigger some events, that are passed to my callback onEvent()-method, right?

Yes. If we receive a WebSocket failure, we propagate it to you via the `WebSocketCallback`’s `onEvent` method you would implement.

`TechnicalEvent.Failure` is the one you’re looking for then:

https://github.com/andregasser/bigbone/blob/1750db8c50b1a47587d554220d20b0fb1baece2b/bigbone/src/main/kotlin/social/bigbone/api/entity/streaming/WebSocketEvent.kt#L42-L47

As written in its kDoc, that’s a terminal event that would close the WebSocket connection. So if you receive it, you’d need to try to connect again if you want to keep the connection going.

> So where should be no unhealthy connection without a trigger?

From what we’ve seen so far: Correct 😊 As long as you do not receive any `TechnicalEvent` such as `TechnicalEvent.Failure`, `TechnicalEvent.Closing` or `TechnicalEvent.Closed`, you should continue receiving events propagated from the Mastodon server.

Hope that helps?
